### PR TITLE
Make fortune file selection random

### PR DESCRIPTION
**Pull request by condret**
Prior to this change, fortune file selection is not not even pseudo-random

- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/core/fortune.c [hunk 1]
```diff
@@ -71,6 +71,7 @@ R_API void r_core_fortune_list_types(void) {
 R_API void r_core_fortune_list(RCore *core) {
 	// TODO: use file.fortunes // can be dangerous in sandbox mode
 	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+
```
Comment (trufae) [inline libr/core/fortune.c:74]:
```suggestion
```

File: libr/core/fortune.c [hunk 2]
```diff
@@ -94,16 +94,27 @@ R_API void r_core_fortune_list(RCore *core) {
...
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		(char *)r_config_get (core->config, "cfg.fortunes.type"), ",", false);
+	if (!r_list_length (types)) {
+		r_list_free (types);
+		return NULL;
+	}
+	char *type = r_str_new ((char *)r_list_get_n (types, r_num_rand (r_list_length (types))));
```
Comment (trufae) [inline libr/core/fortune.c:103]: assuming that cant return null

File: libr/core/fortune.c [hunk 3]
```diff
@@ -94,16 +94,27 @@ R_API void r_core_fortune_list(RCore *core) {
 }
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		(char *)r_config_get (core->config, "cfg.fortunes.type"), ",", false);
+	if (!r_list_length (types)) {
```
Comment (trufae) [inline libr/core/fortune.c:99]: maybe makes sense to have an alias named 'all' to select all fortunefiles (maybe in another pr)

**Review by trufae — APPROVED** [review]

File: libr/core/fortune.c (hunk 2 above)
Comment (condret) [reply libr/core/fortune.c:104]: what can't return NULL?

File: libr/core/fortune.c (hunk 3 above)
Comment (condret) [reply libr/core/fortune.c:100]: should be * imo

File: libr/core/fortune.c (hunk 2 above)
Comment (trufae) [reply libr/core/fortune.c:104]: List.getN can return null if passing the wrong index.. which shouldnt happen.. but if the array is empty the index is 0 or -1 and that will fail i think. So better null check imho

File: libr/core/fortune.c (hunk 3 above)
Comment (trufae) [reply libr/core/fortune.c:100]: yeah, * sounds good. no conflicts with flienames

File: libr/core/fortune.c (hunk 2 above)
Comment (trufae) [reply libr/core/fortune.c:104]: fix this and ok to merge

File: libr/core/fortune.c [hunk 4]
```diff
@@ -94,16 +95,27 @@ R_API void r_core_fortune_list(RCore *core) {
 }
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		(char *)r_config_get (core->config, "cfg.fortunes.type"), ",", false);
```
Comment (trufae) [inline libr/core/fortune.c:99]: why that cast? duplist takes const char*

File: libr/core/fortune.c [hunk 5]
```diff
@@ -94,16 +95,27 @@ R_API void r_core_fortune_list(RCore *core) {
...
+	if (!r_list_length (types)) {
```
Comment (trufae) [inline libr/core/fortune.c:100]: length is a number not a boolean or a pointer, so btter use == 0

File: libr/core/fortune.c [hunk 6]
```diff
@@ -94,16 +95,27 @@ R_API void r_core_fortune_list(RCore *core) {
...
+		r_list_free (types);
+		return NULL;
+	}
+	char *type = r_str_new ((char *)r_list_get_n (types, r_num_rand (r_list_length (types))));
```
Comment (trufae) [inline libr/core/fortune.c:104]:
```suggestion
	const char *file = (char *)r_list_get_n (types, r_num_rand (r_list_length (types)));
	char *type = strdup (file);
```

File: libr/core/fortune.c [hunk 7]
```diff
@@ -94,16 +95,28 @@ R_API void r_core_fortune_list(RCore *core) {
 }
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		(char *)r_config_get (core->config, "cfg.fortunes.type"), ",", false);
```
Comment (trufae) [inline libr/core/fortune.c:99]:
```suggestion
		r_config_get (core->config, "cfg.fortunes.type"), ",", false);
```

File: libr/core/fortune.c [hunk 8]
```diff
@@ -94,16 +95,28 @@ R_API void r_core_fortune_list(RCore *core) {
...
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		r_config_get (core->config, "cfg.fortunes.type"), ",", false);
+	if (!r_list_length (types)) {
+		r_list_free (types);
+		return NULL;
+	}
+	const char *file = (char *)r_list_get_n (types, r_num_rand (r_list_length (types)));
```
Comment (trufae) [inline libr/core/fortune.c:104]:
```suggestion
	const char *file = (const char *)r_list_get_n (types, r_num_rand (r_list_length (types)));
```

File: libr/core/fortune.c [hunk 9]
```diff
@@ -94,16 +95,28 @@ R_API void r_core_fortune_list(RCore *core) {
 }
 
 static char *getrandomline(RCore *core) {
-	const char *types = (char *)r_config_get (core->config, "cfg.fortunes.type");
+	RList *types = r_str_split_duplist (
+		r_config_get (core->config, "cfg.fortunes.type"), ",", false);
+	if (!r_list_length (types)) {
```
Comment (trufae) [inline libr/core/fortune.c:100]:
```suggestion
	if (r_list_empty (types)) {
```

**Comment by trufae** [comment]
fixed all the things. should be mergeable now

**Pull request merged**